examples/developer-tools/src/output-generators/factory_base.ts

```diff
@@ -123,3 +127,75 @@ jsonDefinitionGenerator.forBlock['factory_base'] = function (
 };
 
 jsonDefinitionGenerator.forBlock['text'] = javascriptGenerator.forBlock['text'];
+
+javascriptDefinitionGenerator.forBlock['factory_base'] = function (
+  block: Blockly.Block,
+  generator: JavascriptDefinitionGenerator,
+) {
+  // TODO: Get a JavaScript-legal name for the block
+  const blockName = block.getFieldValue('NAME');
+  const inputsValue = generator.statementToCode(block, 'INPUTS');
+  const inputs = inputsValue
+    ? generator.prefixLines(inputsValue, generator.INDENT) + '\n'
+    : '';
+
+  const alignInputsValue = block.getFieldValue('INLINE');
+  let alignInputs = '';
+  if (alignInputsValue === 'EXT') {
+    alignInputs = 'this.setInputsInline(false)';
+  } else if (alignInputsValue === 'INT') {
+    alignInputs = 'this.setInputsInline(true)';
+  }
+
+  const createConnectionChecks = function (
+    inputName: string,
+    connectionName: string,
+  ): string {
+    if (block.getInput(inputName)) {
+      const check = generator.valueToCode(
+        block,
+        inputName,
+        JsOrder.FUNCTION_CALL,
+      );
+      return `this.set${connectionName}(true, ${check});`;
+    }
+    return '';
+  };
+  const connectionsList = [];
+  connectionsList.push(createConnectionChecks('OUTPUTCHECK', 'Output'));
+  connectionsList.push(createConnectionChecks('TOPCHECK', 'PreviousStatement'));
+  connectionsList.push(createConnectionChecks('BOTTOMCHECK', 'NextStatement'));
+  const connections = connectionsList
+    .filter((value) => value !== '')
+    .join('\n');
+
+  const tooltip = `this.setTooltip(${
+    generator.valueToCode(block, 'TOOLTIP', JsOrder.ATOMIC) || "''"
+  });`;
+  const helpUrl = `this.setHelpUrl(${
+    generator.valueToCode(block, 'HELPURL', JsOrder.ATOMIC) || "''"
+  });`;
+
+  const colourValue = generator.valueToCode(block, 'COLOUR', JsOrder.ATOMIC);
+  const colour = colourValue !== '' ? `this.setColour(${colourValue});` : '';
+
+  // Filter out empty code string pieces, join them all with newlines, and indent them twice.
+  // Inputs code is already indented once automatically by the generator
+  // since it's in a statement input, so it's not included here.
+  const codeStringPieces = generator.prefixLines(
+    [alignInputs, connections, tooltip, helpUrl, colour]
+      .filter((value) => value !== '')
+      .join('\n'),
+    generator.INDENT + generator.INDENT,
+  );
```

Comment:
I wonder if this function might be easier to understand if all the thingies (`alignInputs`, `connections`, `tooltip`, etc.) were created as far as possible in a single template literal, which was then passed to `.prefixLines` directly (or after a split / filter / join, if the blank line filtering can't be done earlier).

e.g. something like:

```JS
  const stringCodePieces = generator.prefixLines(
    `this.setInputsInLine(${ /* .. */ });
${connections}
this.setTooltip(${ /* ... */ });
this.setHelpUrl(${ /* ... */ });`.split('\n').filter(/* ... */).join('\n'));
```